server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -140,7 +140,13 @@ public void postParse(DocumentParserContext context) throws IOException {
 
         // SortedMap is expected to be sorted by key (field name)
         SortedMap<String, BytesReference> dimensionFields = context.doc().getDimensionBytes();
-        if (dimensionFields.isEmpty()) {
+        BytesReference timeSeriesId = buildTsidField(dimensionFields);
+        assert timeSeriesId != null : "In time series mode _tsid cannot be null";
```

Comment:
I don't think the method can really return null anyway. Maybe not worth checking.